Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2592 -- Return a better error message when search times out

Resolves [EREGCSC-2592](https://jiraent.cms.gov/browse/EREGCSC-2592)

**Description**

It's confusing if the search times out and gives 0 results. Let's tell the reader what's actually happening, so that they know to look elsewhere for the information they need.

See Figma comps here:

[Search Page](https://www.figma.com/file/iZFh6UhvGVEHw1D4IAt3Bw/eRegs-Combined-Search?node-id=4652%3A7064&mode=dev)

[Subjects Page](https://www.figma.com/file/KiWw729Nct6tY9gzmvEkVO/Browse-by-Subject?node-id=818%3A16156&mode=dev)

**This pull request changes:**

- Adds new `throwGenericError` method that returns an error after two seconds
   - This method is being used for validation purposes -- initially, all Search and Subjects page requests will fail on purpose to validate the different error message states
- Creates `SearchErrorMsg` Vue component to be used in Search and Subjects page
   - Built to handle states that include a `searchQuery` and states that don't
- Passes `SURVEY_URL` from Django template into Subjects and Search Single Page Apps to be used with error message  feedback link
- Styles message per Figma comps (see above)
- Adds Vitest tests

**Steps to manually verify this change:**

1. First, visit [experimental deployment](https://zvd7igguq5.execute-api.us-east-1.amazonaws.com/dev1224) with `throwGenericError` enabled for all Search and Subjects page searches.  Compare error message to states shown in Figma comps
2. Once `throwGenericError` is disabled, browse around experimental deployment and make sure things behave as expected
